content_cafe.py

```diff
@@ -354,48 +351,49 @@ class ContentCafeSOAPError(IOError):
 class ContentCafeSOAPClient(object):
     """Get historical sales information from Content Cafe through
     its SOAP interface.
-
-    NOTE: This class currently has no test coverage. It wouldn't be too
-    difficult to add coverage for estimate_popularity, at least.
     """
 
     WSDL_URL = "http://contentcafe2.btol.com/ContentCafe/ContentCafe.asmx?WSDL"
 
     DEMAND_HISTORY = "DemandHistoryDetail"
 
-    def __init__(self, user_id, password, wsdl_url=None):
+    def __init__(self, user_id, password, wsdl_url=None, transport=None):
         wsdl_url = wsdl_url or self.WSDL_URL
         self.user_id=user_id
         self.password = password
-        self.soap = SudsClient(wsdl_url)
+        kwargs = dict()
+        if transport:
+            kwargs['transport'] = transport
+        self.soap = SOAPClient(wsdl_url, **kwargs)
 
     def get_content(self, key, content):
         data = self.soap.service.Single(
             userID=self.user_id, password=self.password,
-            key=key, content=content)
-        if hasattr(data, 'Error'):
+            key=key, content=content
+        )
+        if data.Error:
             raise ContentCafeSOAPError(data.Error)
         else:
             return data
 
-    def estimated_popularity(self, key, cutoff=None):
+    def estimated_popularity(self, key, cutoff=None, as_of=None):
```

Comment:
The popularity estimation algorithm depends on the current date, which obviously can't be used in a stable test. This lets us test what value the algorithm _would_ give as of different dates.